docs/source/reference/benchmark/callback.rst

```diff
@@ -0,0 +1,130 @@
+    from sky_callback import step_iterator
+    for batch in step_iterator(train_dataloader):
+        ...
+
+    # Method 2: wrap your loop body with the `step` context manager.
+    for batch in train_dataloader:
+        with sky_callback.step():
+            ...
+
+    # Method 3: call `step_begin` and `step_end` directly.
+    for batch in train_dataloader:
+        sky_callback.step_begin()
+        ...
+        sky_callback.step_end()
+
+That's it.
+Now you can launch your task and get a complete benchmark report using SkyPilot Benchmark CLI.
+`Here <https://github.com/skypilot-org/skypilot/blob/master/examples/benchmark/timm/callback.patch>`_ we provide an example of applying SkyCallback to Pytorch ImageNet training code.
+
+.. note::
+    In distributed training, ``global_rank`` should be additionally passed to ``sky_callback.init`` as follows:
+
+    .. code-block:: python
+
+        # PyTorch DDP users
+        global_rank = torch.distributed.get_rank()
+
+        # Horovod users
+        global_rank = hvd.rank()
+
+        sky_callback.init(
+            global_rank=global_rank,
+            total_steps=num_epochs * len(train_dataloader), # Optional
+        )
+
+Integrations with ML frameworks
+----------------------------------------------------------
+
+Using SkyCallback is even easier for **Keras, PytorchLightning, and HuggingFace Transformers** programs where trainer APIs are used.
+SkyCallback natively supports these frameworks with simple interface.
+
+* Keras example
+
+.. code-block:: python
+
+    from sky_callback import SkyKerasCallback
+
+    # Add the callback to your Keras model.
+    model.fit(..., callbacks=[SkyKerasCallback()])
+
+`Here <https://github.com/skypilot-org/skypilot/blob/master/examples/benchmark/keras_asr/callback.patch>`_ you can find an example of applying SkyCallback to Keras ASR model training.
+
+* PytorchLightning example
+
+.. code-block:: python
+
+    from sky_callback import SkyLightningCallback
+
+    # Add the callback to your trainer.
+    trainer = pl.Trainer(..., callbacks=[SkyLightningCallback()])
+
+`Here <https://github.com/skypilot-org/skypilot/blob/master/examples/benchmark/lightning_gan/callback.patch>`_ you can find an example of applying SkyCallback to PyTorchLightning GAN model training.
+
+* HuggingFace Transformers example
+
+.. code-block:: python
+
+    from sky_callback import SkyTransformersCallback
+
+    # Add the callback to your trainer.
+    trainer = transformers.Trainer(..., callbacks=[SkyTransformersCallback()])
+
+`Here <https://github.com/skypilot-org/skypilot/blob/master/examples/benchmark/transformers_qa/callback.patch>`_ you can find an example of applying SkyCallback to HuggingFace BERT fine-tuning.
+
+.. note::
+    When using the framework-integrated callbacks, do not call ``sky_callback.init`` for initialization.
+    The callbacks will do it for you.
+
+.. SkyCallback for containerized jobs (Advanced)
+.. --------------------------------------------
```

Comment:
Mark as TODO?

Reply:
Removed. I will add this to the doc when we have a user need.